darshan-runtime/lib/darshan-mpiio.c

```diff
@@ -926,6 +930,28 @@ int DARSHAN_DECL(MPI_File_iread)(MPI_File fh, void * buf, int count, MPI_Datatyp
 DARSHAN_WRAPPER_MAP(PMPI_File_iread, int, (MPI_File fh, void * buf, int count, MPI_Datatype datatype, __D_MPI_REQUEST * request),
         MPI_File_iread)
 
+int DARSHAN_DECL(MPI_File_iread_all)(MPI_File fh, void * buf, int count, MPI_Datatype datatype, __D_MPI_REQUEST * request)
+{
+    int ret;
+    double tm1, tm2;
+    MPI_Offset offset;
+
+    MAP_OR_FAIL(PMPI_File_iread_all);
+
+    MPI_File_get_position_shared(fh, &offset);
```

Comment:
Should we be using the regular (non-shared) `MPI_File_get_position()` here?  It look like this function operates on an individual file pointer rather than a shared file pointer per https://www.mpich.org/static/docs/v3.2/www3/MPI_File_iread_all.html .

If my interpretation is right, then we probably need to check all of the wrappers that use those functions (not just the new ones) to make sure we are using the regular version or the shared version in the right places.